aptos-move/move-examples/token_objects/guild/sources/guild.move

```diff
@@ -0,0 +1,309 @@
+/// This module implements the guild token (non-fungible token) including the
+/// functions create the collection and the guild tokens.
+module token_objects::guild {
+    use std::option;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+    use aptos_framework::object::{Self, Object};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::token;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the admin
+    const ENOT_ADMIN: u64 = 2;
+    /// The provided signer is not the owner
+    const ENOT_OWNER: u64 = 3;
+    /// The provided signer is not a guild master
+    const ENOT_GUILD_MASTER: u64 = 4;
+
+    /// The guild token collection name
+    const GUILD_COLLECTION_NAME: vector<u8> = b"Guild Collection Name";
+    /// The guild collection description
+    const GUILD_COLLECTION_DESCRIPTION: vector<u8> = b"Guild Collection Description";
+    /// The guild collection URI
+    const GUILD_COLLECTION_URI: vector<u8> = b"https://guild.collection.uri";
+
+    /// Whitelisted guild masters
+    struct Config has key {
+        whitelist: vector<address>,
```

Comment:
This should use SmartVector instead of vector if the list of whitelisted guild masters can get large

Reply:
Ok. I changed `vector` to `SmartVector` here. When should one use `SmartTable` vs. `SmartVector`?

Reply:
SmartTable/SmartVector should be used when you expect to be a lot of items for better gas efficiency.